include/slam_toolbox/merge_maps_kinetic.hpp

```diff
@@ -64,21 +58,46 @@ class MergeMapTool
   ~MergeMapTool();
 
 private:
+  typedef std::unique_ptr<karto::Pose2> Pose2_ptr;
+  typedef std::unique_ptr<karto::Vector2<kt_double>> Vector2_double_ptr;
   // setup
   void SetConfigs();
 
   // callbacks
   bool MergeMapCallback(slam_toolbox::MergeMaps::Request  &req, slam_toolbox::MergeMaps::Response &resp);
   bool AddSubmapCallback(slam_toolbox::AddSubmap::Request  &req, slam_toolbox::AddSubmap::Response &resp);
   void ProcessInteractiveFeedback(const visualization_msgs::InteractiveMarkerFeedbackConstPtr& feedback);
-  void KartoToROSOccupancyGrid(const karto::LocalizedRangeScanVector& scans);
+  void KartoToROSOccupancyGrid(const karto::LocalizedRangeScanVector& scans, nav_msgs::GetMap::Response& map);
+  void CreateNewMapper();
+  void LoadDatasetFromFile(const std::string& filename);
+
+  Pose2_ptr ApplyCorrection(const karto::Pose2& pose, const tf::Transform& submap_correction)
+  {
+    tf::Transform pose_tf, pose_corr;
+    pose_tf.setOrigin(tf::Vector3(pose.GetX(), pose.GetY(), 0.));
+    pose_tf.setRotation(tf::createQuaternionFromRPY(0, 0, pose.GetHeading()));
+    pose_corr = submap_correction*pose_tf;
+    return Pose2_ptr(new karto::Pose2(pose_corr.getOrigin().x(), pose_corr.getOrigin().y(), tf::getYaw(pose_corr.getRotation())));
+  }
+
+  Vector2_double_ptr ApplyCorrection(const karto::Vector2<kt_double>&  pose, const tf::Transform& submap_correction)
```

Comment:
See above

Reply:
Also what do these functions "functionally" do? The name of the function isnt descriptive enough